web/ui/module/codemirror-promql/src/complete/hybrid.ts

```diff
@@ -191,11 +190,16 @@ export function computeStartCompletePosition(node: SyntaxNode, pos: number): num
     start++;
   } else if (
     node.type.id === OffsetExpr ||
-    (node.type.id === NumberLiteral && node.parent?.type.id === 0 && node.parent.parent?.type.id === SubqueryExpr) ||
+      // Since duration and number are equivalent, writing go[5] or go[5d] is syntactically accurate.
+      // Before we were able to guess when we had to autocomplete the duration later based on the error node,
+      // which is not possible anymore.
+      // So we have to analyze the string about the current node to see if the duration later is already present or not.
+      (node.type.id === AnotherKindOfDuration && !durationTerms.map(v => v.label).includes(currentText[currentText.length - 1])) ||
```

Comment:
my workaround to stop autocomplete duration when there is already a duration syntax used